cmd/controller/controllerspc.go

```diff
@@ -0,0 +1,394 @@
+		}
+		// Finally, if no error occurs we Forget this item so it does not
+		// get queued again until another change happens.
+		c.workqueue.Forget(obj)
+		glog.Infof("Successfully synced '%s'", q.key)
+		return nil
+	}(obj)
+
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+
+	return true
+}
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the spcUpdated resource
+// with the current status of the resource.
+func (c *Controller) syncHandler(key, operation string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the spcUpdated resource with this namespace/name
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	//spcUpdated, err := c.spcLister.StoragePoolClaims(namespace).Get(name)
+	spcUpdated, err := c.clientset.NdmV1alpha1().StoragePoolClaims().Get(name, metav1.GetOptions{})
+	if err != nil {
+		// The spc resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("spcUpdated '%s' in work queue no longer exists", key))
+			return nil
+		}
+
+		return err
+	}
+
+	var sp apis.StoragePool
+	flag := IsDiskAvailable(spcUpdated.Spec.Name)
+	if flag == false {
+		runtime.HandleError(fmt.Errorf("Disk not available: %s", spcUpdated.Spec.Name))
+		return nil
+	}
+
+	switch operation {
+	case "add":
+		sp = DiskOperations(spcUpdated, namespace)
+		Createsp(sp, c, namespace)
+		break
+	case "update":
+		//IsDiskBeingUsed needs to be implemeneted
+		//IsDiskUnusedMounted needs to be implemeneted
+		sp = DiskOperations(spcUpdated, namespace)
+		Updatesp(sp, c, namespace)
+		break
+	case "delete":
+		//IsDiskBeingUsed needs to be implemeneted
+		//IsDiskUnusedMounted needs to be implemeneted
+		err := block.UnMount(spcUpdated.Spec.Name)
+		if err != nil {
+			runtime.HandleError(fmt.Errorf("Unable to unmount ", err))
+			break
+		}
+		Deletesp(spcUpdated.Spec.Name, c, namespace)
+		break
+	}
+
+	return nil
+}
+
+// enqueueSpc takes a spc resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than spc.
+func (c *Controller) enqueueSpc(obj interface{}, q QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+func GetNodeName() (string, error) {
+	var Nodename string
+	NodenameByte, err := exec.Command("uname", "-n").Output()
+	Nodename = string(NodenameByte)
+	return Nodename, err
+}
+
+func DiskOperations(spc *apis.StoragePoolClaim, namespace string) apis.StoragePool {
+	Message := ""
```

Comment:
I am not sure if this kind of error handling helps? e.g. is mount supposed to work if format has issues?